tests/parsing/DefaultArgParserTest.py

```diff
@@ -31,3 +43,25 @@ def test_metavar_not_in_optional_args_sections(self):
                           flags=re.DOTALL)
         self.assertIsNotNone(match)
         self.assertEqual(match.group(1), '-a, --all')
+
+
+class AutocompleteTest(unittest.TestCase):
+
+    def setUp(self):
+        self._old_argcomplete = coalib.parsing.DefaultArgParser.argcomplete
+
+    def tearDown(self):
+        coalib.parsing.DefaultArgParser.argcomplete = self._old_argcomplete
+
+    def test_argcomplete_bear_names(self):
+        self.assertEqual(_argcomplete_bears_names(), get_all_bears_names())
+
+    def test_missing_argcomplete(self):
+        if coalib.parsing.DefaultArgParser.argcomplete is not None:
+            coalib.parsing.DefaultArgParser.argcomplete = None
+        fake_importer = Mock(side_effect=ImportError('import missing'))
+        with patch('builtins.__import__', new=fake_importer):
+            default_arg_parser()
+        self.assertFalse(coalib.parsing.DefaultArgParser.argcomplete)
+        # Dummy usage
+        coalib.parsing.DefaultArgParser.argcomplete = argcomplete
```

Comment:
everything seems legit except for this line. what's this for?

Reply:
I mean when your tear down function is already resetting it

Reply:
It uses the `argcomplete` import at the top of the file which ensures that the test module fails loudly with a suitable error message if it is not installed, instead of the import rather silently failing and the test method being a bit confusing in how it fails.